lesson/narrative_and_checkpoints/narrative-and-checkpoints-template.md

```diff
@@ -1,51 +1,417 @@
+
+    Hint:
+    ```js
+     toEqual(expected)
+     ```
+3. Checkpoint: _Run test in terminal_
+
+    Hint: _Run ```npm test```_
+
+## Exercise 5: _Matcher Functions_
 
 ### Narrative:
 
+Now that we have written our first complete unit test, let us break down five more matchers. We will analyze a test written to test the ```sumList()``` function whose expected behavior is to take in an array of numbers and to output a single number representing the summation of the array items. To test  the ```sumList()``` function we will use five matchers: 
+```js
+//file: potassium_counter.test.js
+const potassiumCounter = require('./potassium_counter')
+ 
+test('add all potassium', () => {
+   //setup
+   const inputList = [1, 2, 3]
+   const expectedValue = 6
+   const value = potassiumCounter.sumList(inputList)
+   //Jest Test Logic
+   expect(value).toBe(expectedValue)
+   expect(value).not.toContain(6)
+   expect(value).toBeTruthy()
+   expect(value).toBeDefined()
+})
+```
+Matchers Used:
+
+1. **.toBe()** is similar to .toEqual() but is used to compare primitive values.
+2. **.not** is used in addition to another matcher to verify that the opposite result is true
+3. **.toContain()** is used when we want to verify that an item is in an array
+4. **.toBeTruthy()** is used to provide a boolean response to the verification of whether a value is truthy or not 
+5. **.toBeDefined()** is used to verify that a variable is not ```undefined``` 
+
+As mentioned in the previous lesson, there are many different matchers and the complete list can be found in the [Jest documentation](https://jestjs.io/docs/expect).
+
+
 ### Instructions:
 
-1. Checkpoint: _Insert checkpoint text here._
+1. Checkpoint: _Rewrite the test switching the matcher with one of the five matchers that we covered, to test that the value should be 6._
+
+    Hint: _Try replacing ```.toEqual()``` with ```.toBe()```_
+
+2. Checkpoint: _Run test in terminal_
+
+    Hint: _Run ```npm test```_
+
+## Exercise 6: _Testing Async Code in Jest (Part 1)_
+
+### Narrative:
+
+Armed with how to write simple unit tests, we will now add a layer of complexity to our tests. We will explore testing asynchronous code in Jest. When testing async code in Jest, we have to be aware of "false positives". False positives occur when a test passes before waiting for a callback to resolve. By default, Jest is not aware that it must wait for callbacks to resolve before finishing a test. Let's test the following function that calls to a REST API and returns a callback with the data:
+```js
+//file: potassium_counter.test.js
+const potassiumCounter = require('./potassium_counter')
+ 
+test('successfully request potassium level and return in callback', () => {
+   //setup
+  const inputFood = null
+  const value = potassiumCounter.potassiumSearch(inputFood, result => {
+       //Jest test
+        expect(result).toBeDefined()
+      })
+})
```

Comment:
This is not how to test async code. What we're doing here is mocking the callback function. The test code shouldn't be used this way.

Reply:
This seems in line with the examples in the Jest docs. Is there a distinction that you're noting that I'm missing?

https://jestjs.io/docs/asynchronous